server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -218,18 +216,7 @@ private void refreshExistingFileStateIfNeeded(ClusterState clusterState) {
     }
 
     public boolean watching() {
-        return this.watchService != null;
-    }
-
-    private void cleanupWatchKeys() {
-        if (settingsDirWatchKey != null) {
-            settingsDirWatchKey.cancel();
-            settingsDirWatchKey = null;
-        }
-        if (configDirWatchKey != null) {
-            configDirWatchKey.cancel();
-            configDirWatchKey = null;
-        }
+        return watcherThread != null;
```

Comment:
Sorry I missed this last time, is there any reason why we changed this bit here? I think it's not fully correct, because we are sort of watching what's in the file directory on startup before we start the watcher thread.

Reply:
My logic here was that we are actively watching when there's a thread processing the events, but it's all basically the same at this point. I'll change it back.

Reply:
Ah I see the dependency in stop now, it's good, we should keep it as you have it.